### PR TITLE
prep ad hoc Schematron for mass validation

### PR DESCRIPTION
**Pull request by sydb**
This PR is an attempt to update our ad hoc Schematron schema (`common/schema/dhqTEI-ready.sch`) in preparation for using it in mass validation. This primarily meant removing the case of a rule from one pattern extending an abstract rule ("target-uri-constraints") defined in another pattern. I also made the test for PIs slightly more robust, the test for duplicate `@xml:id` easier to read, and re-wrote some comments.

I have tested these changes thoroughly by validating all the `articles/0*[012]/*.xml` files (i.e., roughly 30% of all the `articles/0*/*.xml` files) against this schema both before and after the changes (using schxslt-1.9.5/2.0/pipeline-for-svrl.xsl). The results were _exactly_ the same. Thus I think there is no harm whatsoever in merging this PR. So much so I think only 0 or 1 reviewer need approve before it is merged.

HOWEVER, it is worth noting that when the validation was performed against the altered version of the schema using schxslt2-1.4.4/transpile.xsl instead (which cannot be used on the original schema, hence the desire to update it), the results were _drastically_ different. Hundreds of messages that were not present before. This fact has no direct bearing on _this PR_, but does mean that updating to a modern Schematron processor (namely SchXslt2) might be a much bigger deal than I initially thought.

### CODE REVIEW

File: common/schema/dhqTEI-ready.sch [hunk 1]
```diff
@@ -1,346 +1,387 @@
...
+
+
+  <pattern id="top-level">
+    <!-- Pointing to a schema on the internet means the file would not
+         be portable. Thus check that the references to schemas in the
+         <?oxygen?> or <?xml-model?> processing instructions (on their
+         RNGschema=, SCHschema, or href= pseudo attributes) do not
+         start with "http". -->
+    <rule context="/processing-instruction()">
+      <report test="matches(.,'(RNGSchema|SCHSchema|href)=\s*.http')" role="warning">
```
[the remark this answers is not in the document]
Comment (sydb) [reply common/schema/dhqTEI-ready.sch:52]: We are not worried _at all_ that there are any spaces before or after ‘=’ in a processing instruction in an article file _right now_. What we perhaps should worry about is someone accidentally inserting such a space in the future. Seems to me that should be caught by the schema, rather than the code, though, no?
(“Then why did you test for a space after the ‘=’ sign, Syd?”. I have no idea, I probably typed it before I thought about it. :-)

P.S. Worth noting that on my list of things to do is to convert all those "oxygen" processing instructions to proper "xml-model" PIs. That would be a good time to add a schema check for whitespace within the pseudo-attributes.
